File: backend/lib/newspaper_scraper/model/article_summary.ex
defmodule NewspaperScraper.Model.ArticleSummary do
  @moduledoc """
  This module defines the struct for article summaries
  """

  alias NewspaperScraper.Model.Author

  @type t :: %__MODULE__{
          newspaper: String.t(),
          authors: list(Author.t()),
          title: String.t(),
          excerpt: String.t(),
          date_time: String.t(),
          url: String.t(),
          is_premium: boolean()
        }

  @art_sum_keys [
    :newspaper,
    :authors,
    :title,
    :excerpt,
    :date_time,
    :url,
    :is_premium
  ]

  @derive {Jason.Encoder, only: @art_sum_keys}
  @enforce_keys @art_sum_keys
  defstruct @art_sum_keys
end

File: frontend/src/types/scraper/articleSummary.ts
import type { Author } from './author'

/**
 * Represents the summary of an article
 * @interface ArticleSummary
 */
export interface ArticleSummary {
    /**
     * The newspaper name
     * @type {string}
     */
    newspaper: string
    /**
     * The article's authors
     * @type {Author[]}
     */
    authors?: Author[]
    /**
     * The article's title
     * @type {string}
     */
    title: string
    /**
     * The article's excerpt
     * @type {string}
     */
    excerpt: string
    /**
     * The article's date and time in locale format
     * @type {string}
     */
    date_time: string
    /**
     * The article's url
     * @type {string}
     */
    url: string
    /**
     * Indicates whether the article is premium or not
     * @type {string}
     */
    is_premium: boolean
}
